main/acle.md

```diff
@@ -7766,6 +7920,1419 @@ This is because the ACLE intrinsic calls do not imply a particular
+specified in this section.
+
+### SME PSTATE functions
+
+#### Prototypes
+
+``` c
+  __attribute__((arm_streaming_compatible))
+  bool __arm_in_streaming_mode(void);
+
+  // Function with external linkage.
+  __attribute__((arm_streaming_compatible))
+  void __arm_za_disable(void);
+```
+
+#### Semantics
+
+<!-- Do not remove the following `span`, they are needed to create an
+anchor that can be referred via an internal hyperlink to the paragraph
+following it. --><span id="__arm_in_streaming_mode"></span>
+
+**`__arm_in_streaming_mode()`**
+
+> This call returns true if the [abstract machine](#abstract-machine) is
+> in streaming mode. It always returns true when called from [streaming
+> statements](#streaming-statement) and it always return false when
+> called from [non-streaming statements](#non-streaming-statement).
+> However, the call is not semantically a constant expression even in
+> those cases.
+
+<!-- Do not remove the following `span`, they are needed to create an
+anchor that can be referred via an internal hyperlink to the paragraph
+following it. --><span id="__arm_za_disable"></span>
+
+**`__arm_za_disable()`**
+
+> This call commits any pending lazy save and turns ZA off;
+> see [[AAPCS64]](#AAPCS64) for details.
+>
+> Note: since this is a [private-ZA](#private-za) function,
+> it is valid (though pointless) to call it from a function that has
+> [ZA state](#za-state). The compiler would simply reenable ZA after the
+> call and reload the saved ZA state.
+
+### SME ZA state assertions
+
+#### Prototypes
+
+``` c
+  __attribute__((arm_streaming_compatible, arm_shared_za))
+  void svundef_za();
+```
+
+#### Semantics
+
+**`svundef_za()`**
+
+> This call asserts that ZA does not currently have any useful data.
+> Semantically, it fills ZA with unpredictable data, although from
+> a quality of implementation perspective, it should not generate
+> any object code.
+>
+> The call can be used as an optimization hint to the compiler, so that
+> the compiler does not insert unnecessary code to save and restore the
+> current ZA contents. The call might also be useful for static analysis.
+
+### SME instruction intrinsics
+
+#### Common rules
+
+The intrinsics in this section have the following properties in common:
+
+*   Every argument named `tile`, `slice_offset` or `tile_mask` must
+    be an integer constant expression.
+
+*   ZA loads and stores do not use typed pointers, since there is
+    no C or C++ type information associated with the contents of ZA.
+
+*   Intrinsics have a `_hor` suffix if they operate on horizontal slices
+    of a given ZA tile and a `_ver` suffix if they operate on vertical
+    slices of a given ZA tile.
+
+#### LD1B, LD1H, LD1W, LD1D, LD1Q
+
+``` c
+  // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
+  __attribute__((arm_streaming, arm_shared_za))
+  void svld1_hor_za8(uint64_t tile, uint32_t slice_base,
+                     uint64_t slice_offset, svbool_t pg, const void *ptr);
```

Comment:
I am curious why we need the `uint64_t tile` parameter for the `_za8` version and why we need the `uint64_t slice_offset` for `_za128` version? Is it valid for either cases to take a value other than `0`?

Reply:
Gah, this a pasto from the LD1 functions.  Thanks for the catch.